package/kernel/mac80211/patches/rt2x00/605-wifi-rt2x00-support-loading-eeprom-from-NVMEM-ce.patch

```diff
@@ -0,0 +1,87 @@
+From: Shiji Yang <yangshiji66@outlook.com>
+Date: Tue, 18 Jul 2023 20:18:16 +0800
+Subject: [PATCH] wifi: rt2x00: support loading eeprom from NVMEM cells
+
+This patch allows rt2x00 to load eeprom from "eeprom" NVMEM cell.
+Add "ralink,eeprom-swap" property will perform extra byte swapping
+operation for the eeprom.
+
+Example:
+
+/* load eeprom from NVMEM provider 'eep' */
+&wmac {
+	nvmem-cells = <&eep>;
+	nvmem-cell-names = "eeprom";
+	ralink,eeprom-swap;
+};
+
+Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
+---
+ .../net/wireless/ralink/rt2x00/rt2x00eeprom.c | 44 +++++++++++++++++++
+ 1 file changed, 44 insertions(+)
+
+--- a/drivers/net/wireless/ralink/rt2x00/rt2x00eeprom.c
++++ b/drivers/net/wireless/ralink/rt2x00/rt2x00eeprom.c
+@@ -30,6 +30,7 @@
+ #include <linux/mtd/mtd.h>
+ #include <linux/mtd/partitions.h>
+ #endif
++#include <linux/nvmem-consumer.h>
+ #include <linux/of.h>
+ 
+ #include "rt2x00.h"
+@@ -96,6 +97,44 @@ static int rt2800lib_read_eeprom_mtd(str
+ }
+ #endif
+ 
++static int rt2800lib_read_eeprom_nvmem(struct rt2x00_dev *rt2x00dev)
++{
++	static struct firmware fw;
++	struct device_node *np = rt2x00dev->dev->of_node;
++	unsigned int len = rt2x00dev->ops->eeprom_size;
++	struct nvmem_cell *cell;
++	const void *data;
++	size_t retlen;
++	int i;
++
++	cell = of_nvmem_cell_get(np, "eeprom");
++	if (IS_ERR(cell))
++		return PTR_ERR(cell);
++
++	data = nvmem_cell_read(cell, &retlen);
++	nvmem_cell_put(cell);
++
++	if (IS_ERR(data))
++		return PTR_ERR(data);
++
++	if (retlen != len) {
++		dev_err(rt2x00dev->dev, "invalid eeprom size, required: 0x%04x\n", len);
++		kfree(data);
++		return -EINVAL;
++	}
++
++	rt2x00dev->eeprom = (__le16 *)data;
++
++	if (of_find_property(np, "ralink,eeprom-swap", NULL))
++		for (i = 0; i < len/sizeof(u16); i++)
++			rt2x00dev->eeprom[i] = swab16(rt2x00dev->eeprom[i]);
```

Comment:
Is this actually used? Can't find it used in the current conversion.

Reply:
Notice this same thing is for the mtd... Do we use the mtd properties? 

In any case I would generalize the code and unify the properties... (for the conversion a simple grep and sed should do the trick)

Reply:
> Is this actually used? Can't find it used in the current conversion.

Yes, some lantiq devices need it.
https://git.openwrt.org/?p=openwrt%2Fopenwrt.git&a=search&h=HEAD&st=grep&s=ralink%2Cmtd-eeprom-swap%3B

BTW, I think we can change the mtd eeprom patch property name to the same one `ralink,eeprom-swap`.
https://github.com/openwrt/openwrt/blob/b35756446391d1b21970a0e05f8f6f5e069f73cf/package/kernel/mac80211/patches/rt2x00/604-rt2x00-load-eeprom-on-SoC-from-a-mtd-device-defines-.patch#L85